app/src/main/java/io/appium/uiautomator2/core/AccessibilityNodeInfoHelper.java

```diff
@@ -44,4 +49,43 @@ static Rect getVisibleBoundsInScreen(AccessibilityNodeInfo node, int width, int
         nodeRect.intersect(displayRect);
         return nodeRect;
     }
-}
\ No newline at end of file
+
+    /**
+     * Perform accessibility action ACTION_SET_PROGRESS on the node
+     *
+     * @param value desired progress value
+     * @return true if action performed successfully
+     */
+    public static boolean setProgressValue(final AccessibilityNodeInfo node, final float value) {
+        if (node.getActionList().contains(AccessibilityAction.ACTION_SET_PROGRESS)) {
```

Comment:
if (!node.getActionList().contains(AccessibilityAction.ACTION_SET_PROGRESS)) {
Logger.debug("Element does not support ACTION_SET_PROGRESS action.");
return false;
}
...